python/TestHarness/tests/resultsstore/common.py

```diff
@@ -0,0 +1,349 @@
+# This file is part of the MOOSE framework
+# https://mooseframework.inl.gov
+#
+# All rights reserved, see COPYRIGHT for full restrictions
+# https://github.com/idaholab/moose/blob/master/COPYRIGHT
+#
+# Licensed under LGPL 2.1, please see LICENSE for details
+# https://www.gnu.org/licenses/lgpl-2.1.html
+
+"""Contains common testing utilities for TestHarness.resultsstore."""
+
+import os
+import random
+import string
+from dataclasses import dataclass
+from io import StringIO
+from typing import Any, Iterator, Optional, Tuple
+from unittest import TestCase
+
+from bson.objectid import ObjectId
+from mock import patch
+from pymongo import MongoClient
+from pymongo.database import Database
+from TestHarness.resultsstore.civetstore import CIVETStore
+from TestHarness.resultsstore.utils import TestName
+from TestHarness.tests.common import TestHarnessResultCache
+
+# Dummy APPLICATION_REPO variable from the civet environment
+APPLICATION_REPO = "git@github.com:idaholab/moose"
+# Dummy CIVET_SERVER variable used in the civet environment
+CIVET_SERVER = "https://civet-be.inl.gov"
+
+# Static object id for use in testing
+OBJECT_ID = ObjectId()
+
+
+def random_git_sha() -> str:
+    """Generate a random Git SHA."""
+    hex_characters = string.hexdigits[:16]
+    random_sha = "".join(random.choice(hex_characters) for _ in range(40))
+    return random_sha
+
+
+def random_id() -> int:
+    """Generate a random ID."""
+    return random.randint(1, 1000000)
+
+
+def base_civet_env() -> Tuple[str, dict]:
+    """Build a random base CIVET environment for testing."""
+    # Dummy sha to use for the base commit
+    base_sha = random_git_sha()
+    # Dummy civet environment
+    env = {
+        "APPLICATION_REPO": APPLICATION_REPO,
+        "CIVET_BASE_SHA": random_git_sha(),
+        "CIVET_BASE_SSH_URL": "git@github.com:idaholab/moose.git",
```

Comment:
Would this be valid?
```suggestion
        "CIVET_BASE_SSH_URL": APPLICATION_REPO + ".git",
```